test/e2e/utils.go

```diff
@@ -40,39 +42,70 @@ var (
 	bondConunter  = 0
 )
 
-func writePodsLogs(namespace string, sinceTime time.Time, writer io.Writer) error {
+func writePodsLogs(namespace string, sinceTime time.Time, testFailed bool) error {
 	if framework.Global.LocalOperator {
 		return nil
 	}
+	OpenTestLogFile("pod", func(writer io.ReadWriteCloser){
+		podLogOpts := corev1.PodLogOptions{}
+		podLogOpts.SinceTime = &metav1.Time{sinceTime}
+		podList := &corev1.PodList{}
+		err := framework.Global.Client.List(context.TODO(), podList, &dynclient.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		podsClientset := framework.Global.KubeClient.CoreV1().Pods(namespace)
+
+		for _, pod := range podList.Items {
+			appLabel, hasAppLabel := pod.Labels["app"]
+			if !hasAppLabel || appLabel != "kubernetes-nmstate" {
+				continue
+			}
+			req := podsClientset.GetLogs(pod.Name, &podLogOpts)
+			podLogs, err := req.Stream()
+			if err != nil {
+				io.WriteString(GinkgoWriter, fmt.Sprintf("error in opening stream: %v\n", err))
+				continue
+			}
+			defer podLogs.Close()
+			rawLogs, err := ioutil.ReadAll(podLogs)
+			if err != nil {
+				io.WriteString(GinkgoWriter, fmt.Sprintf("error reading kubernetes-nmstate logs: %v\n", err))
+				continue
+			}
+			formattedLogs := strings.Replace(string(rawLogs), "\\n", "\n", -1)
+			io.WriteString(writer, formattedLogs)
+			// If test failed print pods logs to screen
+			if testFailed{
```

Comment:
Ginkgo is already not printing GinkgoWriter if test fail, but we disable it by passing "ginkgo.v" maybe we want to remove it again.

Reply:
Not sure , to say the truth. There parts of the logs that are needed to be printed

Reply:
Ack